### PR TITLE
SO-5919: Return access token as part of User object on login

### PR DESCRIPTION
**Pull request by AAAlinaaa**

### CODE REVIEW

**Review by cmark — APPROVED** [review]
🥓

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/client/TransportClient.java [hunk 1]
```diff
@@ -226,17 +226,17 @@ public User connect(final String username, final String password) throws Snowowl
...
 				.buildAsync()
 				.execute(bus)
 				.getSync(3L, TimeUnit.SECONDS);
 			
 			// if successfully logged in replace the event bus with an authorized one
-			env.services().registerService(IEventBus.class, new AuthorizedEventBus(bus, ImmutableMap.of("Authorization", token.getToken())));
+			env.services().registerService(IEventBus.class, new AuthorizedEventBus(bus, ImmutableMap.of("Authorization", user.getAccessToken().getToken())));
 			
-			return IdentityProvider.authJWT(token.getToken());
+			return IdentityProvider.authJWT(user.getAccessToken().getToken());
```
Comment (apeteri) [inline core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/client/TransportClient.java]: I thought this was the primary point of the PR:

```suggestion
			return user;
```

`IdentityProvider.authJWT` uses timestamp verification and we don't want to rely on the precision of the client's timekeeping.

**Review by apeteri — CHANGES_REQUESTED** [review]

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/client/TransportClient.java (hunk 1 above)
Comment (cmark) [reply core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/client/TransportClient.java]: It should be yes, @AAAlinaaa pls fix.

**Review by cmark — APPROVED** [review]
Yes, it looks good now.

**Review by apeteri — APPROVED** [review]
🍃

**Pull request merged**